framework/src/postprocessors/InterfaceIntegralVariableValuePostprocessor.C

```diff
@@ -46,13 +46,52 @@ InterfaceIntegralVariableValuePostprocessor::InterfaceIntegralVariableValuePostp
     _grad_u_neighbor(parameters.isParamSetByUser("neighbor_variable")
                          ? coupledNeighborGradient("neighbor_variable")
                          : coupledNeighborGradient("variable")),
-    _interface_value_type(parameters.get<MooseEnum>("interface_value_type"))
+    _interface_value_type(parameters.get<MooseEnum>("interface_value_type")),
+    _neighbor_fv_variable(
+        parameters.isParamSetByUser("neighbor_variable")
+            ? dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("neighbor_variable", 0))
+            : dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("variable", 0))),
+    _fv(_fv_variable)
 {
   addMooseVariableDependency(&mooseVariableField());
+
+  /// Primary and secondary variable should both be finite volume variables or both not be
+  if ((_fv && !getFieldVar("neighbor_variable", 0)->isFV()) ||
+      (!_fv && getFieldVar("neighbor_variable", 0)->isFV()))
+    mooseError("For the InterfaceIntegralVariableValuePostprocessor, if variable "
+               "is a finite volume variable, so should be neighbor_variable.");
```

Comment:
You might just say that the variable types of `variable` and `neighbor_variable` must match or use some other language that isn't finite volume of finite element specific